sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/codegen/BufferHolder.java

```diff
@@ -61,12 +61,15 @@
    * Grows the buffer by at least neededSize and points the row to the buffer.
    */
   void grow(int neededSize) {
-    if (neededSize > ARRAY_MAX - totalSize()) {
+    assert neededSize < 0 :
+      "Cannot grow BufferHolder by size " + neededSize + " because the size is negative";
+    int roundedSize = ByteArrayMethods.roundNumberOfBytesToNearestWord(neededSize);
```

Comment:
shall we do the same thing in L55?

Reply:
and we should add a comment to the `buffer` field to say that, it's guaranteed to be word-aligned, and explain who may assume it's word-aligned.